scylla-cql/src/errors.rs

```diff
@@ -58,6 +59,31 @@ pub enum QueryError {
     RequestTimeout(String),
 }
 
+/// An error type that occurred when executing one of:
+/// - QUERY
+/// - PREPARE
+/// - EXECUTE
+/// - BATCH
+///
+/// requests.
+#[derive(Error, Debug)]
+pub enum UserRequestError {
+    #[error("Database returned an error: {0}, Error message: {1}")]
+    DbError(DbError, String),
+    #[error(transparent)]
+    CqlResultParseError(#[from] CqlResultParseError),
```

Comment:
No, the reason this type was introduced is to filter out the "impossible" error variants from `CqlResponseParseError` (see `From<RequestError> for UserRequestError`). It's just a transition error type (same goes for `RequestError`). Actually, you reminded me that I need to make them private.

Reply:
Yes, if it's not exposed to user then it should not be pub

Reply:
The thing is that almost all error types are contained in `scylla-cql` crate. The types need to be `pub` there if they are going to be used by `scylla`.

Now, I see two solutions:
- move private error types to `scylla` crate (and make them `pub(crate)`
- hide the private error types using re-exports

I'm not sure which one is better. WDYT?

Reply:
I think that all error types that are not used in scylla-cql should be moved to scylla - but probably in a separate PR

Reply:
> No, the reason this type was introduced is to filter out the "impossible" error variants from `CqlResponseParseError` (see `From<RequestError> for UserRequestError`). It's just a transition error type (same goes for `RequestError`). Actually, you reminded me that I need to make them private.

I'm confused. Here you say that it's just a transition type and should be private, but commit message says `This error type will be exposed to users, and returned in case of
failure of user request (requests triggered via public API, i.e.
PREPARE, EXECUTE, QUERY, BATCH).`

So which is it?